lib/guardian.rb

```diff
@@ -110,14 +110,14 @@ def is_whisperer?
   end
 
   def is_category_group_moderator?(category)
-    return false unless category
-    return false unless authenticated?
+    return false if !SiteSetting.enable_category_group_moderation?
+    return false if !category
+    return false if !authenticated?
+    return false if !category.reviewable_by_group_id.present?
+
     @is_category_group_moderator ||= {}
     @is_category_group_moderator[category.id] ||= begin
```

Comment:
1. This doesn't memoize `false` values
2. Now that it's reduced to this single query - it'd be more efficient to key it by `category.reviewable_by_group_id`